### PR TITLE
Implemented another F5 irule management module, using the 'bigsuds' library instead

### PR DESCRIPTION
**Pull request by Rylon**
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- bigip_irule_bigsuds

##### ANSIBLE VERSION

```
ansible 2.2.0.0
```

##### SUMMARY

The reason for this change is that when we used the existing module to manage our irules, we were unable to do so unless the username we provided had full Administrator access. If the user only had 'Manager' permissions, we would receive a `java.lang.SecurityException: Authorization failed` exception from `com.f5.rest.workers.ForwarderWorker.evaluatePermission(ForwarderWorker.java:420)`.

Since we were already successfully using `bigip_virtual_server` (which uses `bigsuds`), we decided to reimplement the irule module to use the same library. This new module has effectively the same interface as the existing `bigip_irule` module.

I'm contributing this in the hopes that it will be useful to others in the same situation, please let me know what you think? Thanks!

### CODE REVIEW

File: network/f5/bigip_irule_bigsuds.py
Comment (gundalow) [inline network/f5/bigip_irule_bigsuds.py]: 2.3

File: network/f5/bigip_irule_bigsuds.py
Comment (gundalow) [inline network/f5/bigip_irule_bigsuds.py:3]: Copyright & license notice missing. You can copy this from one of the existing modules

File: network/f5/bigip_irule_bigsuds.py
Comment (gundalow) [inline network/f5/bigip_irule_bigsuds.py]: To get this formatted correct please write as:
C(src)

File: network/f5/bigip_irule_bigsuds.py
Comment (gundalow) [inline network/f5/bigip_irule_bigsuds.py]: For consistence we don't put a fullstop on `short_description`,
We do however on the `description:` require them to be full sentences.

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by Rylon** [comment]
Thanks @gundalow for the review, I will make the appropriate changes and update the PR shortly :)

**Comment by gundalow** [comment]
@Rylon Thanks for this new module, I've made a few comments on it regarding the documentation. I'll let @caphrim007 (from F5) do a fuller review.

For new (or updates) to Network modules in 2.3 we require integration tests to be added. The F5 tests are currently in a separate repo and not in Ansible yet. @caphrim007 feel free to poke me on IRC and we can discuss how we want to improve this moving forward.

**Comment by Rylon** [comment]
Thanks @gundalow I've updated the PR with those changes and will wait for the full review. 👍

**Comment by caphrim007** [comment]
@gundalow the original intent of the modules that I was intending was to make them based purely off rest, but due to bugs that @Rylon points out which appear to be out of my control, it looks like we will want to support soap as well.

However, I would prefer not to diverge the module names. So in the interest of having a consistent module name for the f5 modules, @Rylon can we instead put SOAP code into the existing irule module and let it determine which method to use?

@Rylon what version of bigip is this happening for you on so that I can repro?

**Comment by Rylon** [comment]
@caphrim007 yeah, I wasn't sure of the best approach there regarding the name, but I agree about not wanting to diverge. The BIG-IP version we encountered this issue with is 11.5.3.

So am I right to say that `f5-sdk` is the preferred mechanism now, as that uses iControl REST API, instead of SOAP?

**Comment by gregdek** [comment]
Thanks @Rylon. To the current maintainers, @mhite, @caphrim007, @gundalow, @qalthos, @privateip please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by caphrim007** [comment]
@Rylon yep, f5-sdk is the preferred way, but its a problem if the rest api is acting like this. So in cases like this, I'm willing to let SOAP slide and have both in the module. 

To the extent possible I'd like to not have to have the user specify their transport preference. I'd like it for the module to decide for them, and then cover testing in the functional and unit tests that are, at the moment, in the f5-ansible repo

**Comment by caphrim007** [comment]
@Rylon can you try this change and see if it makes your manager account work with the existing irule module?

Find the ManagementRoot line and add a parameter to it; `token=True`.

Lemme know what happens

**Comment by Rylon** [comment]
Sure @caphrim007 I'll take a look later today :)

**Comment by Rylon** [comment]
@caphrim007 could you please confirm which module I should make that change in? I tried line 157 of `bigip_rule.py`, but received the same error. Thanks!

**Comment by caphrim007** [comment]
@Rylon it should be after the port= portion of that line.

port=..., token=True)

**Comment by caphrim007** [comment]
@Rylon  yes, bigip_irule. Are you using ldap or tacacs to auth to your bigip?

**Comment by Rylon** [comment]
@caphrim007 Hi, unfortunately I got the same error with that suggested change. We're not using anything like LDAP for user management, so far as I know, it's just the built in user management.

**Comment by caphrim007** [comment]
@Rylon is this a hotfix release? Or just the base version? you can get that from "tmsh show sys version".

**Comment by caphrim007** [comment]
the 11.5.3 bigip you're using that is

**Comment by Rylon** [comment]
@caphrim007 it is indeed a hotfix :)

```
  Product  BIG-IP
  Version  11.5.3
  Build    2.10.196
  Edition  Engineering Hotfix HF2
```

**Comment by caphrim007** [comment]
ok, i'm in the middle of creating a repro env for it to see if I can figure out what's going on. Have you ever uploaded your bigip config to ihealth?

**Comment by caphrim007** [comment]
@Rylon ok, i'm in the middle of creating a repro env for it to see if I can figure out what's going on. Have you ever uploaded your bigip config to ihealth?

**Comment by Rylon** [comment]
@caphrim007 yeah we did in the past, but haven't recently.

**Pull request closed without merging**
